### PR TITLE
fail fast until an after is found

### PR DESCRIPTION
**Pull request by mcollina**
Fixes https://github.com/mcollina/avvio/pull/45
Fixes https://github.com/fastify/fastify/issues/757

cc @fastify @allevo

### CODE REVIEW

**Comment by sarneeh** [comment]
Nice! 👍

**Comment by allevo** [comment]
@mcollina

This is a breaking change. Should we ship this on fastify after its rc?
This is not configurable. Is it a problem? 

Anyway it is fine for me

**Comment by mcollina** [comment]
I see this as a bug, not as a breaking change. This behavior was not really documented before, and I would guess it is what people expects. I will update the docs to reflect it, too.

File: boot.js [hunk 1]
```diff
@@ -191,12 +191,13 @@ Boot.prototype._addPlugin = function (plugin, opts) {
   current.enqueue(obj, (err) => {
     if (err) {
       this._error = err
+      // current.error()
```
Comment (delvedor) [inline boot.js]: What is this?

**Comment by jimmiehansson** [comment]
Great work and fast solution 🎖 🚀 🤖 🌮

**Comment by mcollina** [comment]
@delvedor check this out.

**Comment by mcollina** [comment]
> IMO the best thing is that if a plugin throws and there is not an after just after it then we should throw.
Since now we have this.isAfter it should be easy to achieve.

`.ready(function (err) { })` expects the errors to end up there. I think that is preferred to a uncatchable throw.

File: README.md [hunk 2]
```diff
@@ -191,6 +195,9 @@ The callback changes basing on the parameters your are giving:
 3. If two parameters are given to the callback, the first will be the `error` object, the second will be the `done` callback.
 4. If three parameters are given to the callback, the first will be the `error` object, the second will be the top level `context` unless you have specified both server and override, in that case the `context` will be what the override returns, and the third the `done` callback.
 
+In the no parameter and one parameter variants, the callback can also
```
Comment (jsumners) [inline README.md]: Add quotes (`"`) around `no parameter` and `one parameter`. It'll be easier to read.

**Comment by delvedor** [comment]
> `ready(function (err) { })` expects the errors to end up there. I think that is preferred to a uncatchable throw.

Yeah, what I mean is:

```js
app.use((instance, opts, next) => {
  next(new Error('kaboom'))
})

app.after(err => {
  console.log(err)
})

app.use((instance, opts, next) => {
  next(new Error('kaboom'))
})

app.use((instance, opts, next) => {
  console.log('this should not be called')
})

app.ready(err => {
  console.log(err)
})
```

**Comment by jsumners** [comment]
So that I am clear:

```js
app
  .use((instance, opts, next) => {
    // Do something
    next()
  })
  .use((instance, opts, next) => {
    next(Error('nope'))
  })
  .use((instance, opts, next) => {
    // skipped
  })

app.ready((err) => {
  // `err` is not `undefined`
  // And the error was not thrown due to a missing `.after`
})

**Comment by mcollina** [comment]
@jsumners yes.

The error is never thrown, it ends up in `ready((err) => {})`.

**Comment by jsumners** [comment]
LGTM with that documentation nit.

**Pull request merged**